distributed/cluster.py

```diff
@@ -2,65 +2,196 @@
 
 import paramiko
 from time import sleep
+import socket
+import os
 
-from toolz import assoc
+from toolz import merge
 
+# These are handy for creating colorful terminal output to enhance readability
+# of the output generated by dcluster.
+class bcolors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+def start_center(logdir, center_addr, center_port):
 
-def start_center(addr):
     ssh = paramiko.SSHClient()
     ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-    ssh.connect(addr)
+    ssh.connect(center_addr, timeout = 20)
 
-    channel = ssh.invoke_shell()
-    channel.settimeout(20)
-    sleep(0.1)
-    channel.send('dcenter --host %s\n' % addr)
-    channel.recv(10000)
+    cmd = 'dcenter --host {addr} --port {port}'.format(
+        addr = center_addr, port = center_port, logdir = logdir)
 
-    return {'client': ssh, 'channel': channel}
+    # Optionally re-direct stdout and stderr to a logfile
+    if logdir is not None:
+        cmd = 'mkdir -p {logdir} && ' + cmd
+        cmd += '&> {logdir}/dcenter_{addr}:{port}.log\n'.format(
+            addr = center_addr, port = center_port, logdir = logdir)
```

Comment:
I'm not sure if it makes sense to use the logging module directly to direct logs to a file.  Presumably this would mean adding extra logic into `dcenter/dworker` to change logging levels and add `logging.FileHandlers`.  @quasiben you've played a lot with logging recently.  Is there is any benefit one way or the other?